DiskImager/Disk.cs

```diff
@@ -605,16 +605,16 @@ private Int64 ParseMBRForSize(byte[] buffer)
 
             // Following part is attempt to determine the size of all partitions 
             // by last byte of one of them
-            var endOfLastPartition = 0L;
+            Int64 endOfLastPartition = 0L;
 
             if(mbr.firstPartition.type != EnumPartitionType.EMPTY)
             {
-                endOfLastPartition = (mbr.firstPartition.sectorsFromMBRToPartition + mbr.firstPartition.sectorsInPartition) * 512;
+                endOfLastPartition = (mbr.firstPartition.sectorsFromMBRToPartition + mbr.firstPartition.sectorsInPartition) * 512L;

```

Comment:
I'm thinking that I need to move 512L to constant and check its possible usage in other places.